code/datums/elements/immerse.dm

```diff
@@ -111,7 +111,10 @@
 		return
 	if(HAS_TRAIT(movable, TRAIT_IMMERSED))
 		return
-	if(movable.layer >= ABOVE_ALL_MOB_LAYER || !ISINRANGE(movable.plane, MUTATE_PLANE(FLOOR_PLANE, source), MUTATE_PLANE(GAME_PLANE, source)))
+	if(!ISINRANGE(movable.plane, MUTATE_PLANE(FLOOR_PLANE, source), MUTATE_PLANE(GAME_PLANE, source)))
+		return
+	//First, floor plane objects use TOPDOWN_LAYER, second this check shouldn't apply to them anyway.
+	if(movable.plane > MUTATE_PLANE(FLOOR_PLANE, source) && movable.layer >= ABOVE_ALL_MOB_LAYER)
```

Comment:
do you want to prevent ALL topdown layer usages? this seems like it could break pretty easy in future.
Might be worth making a macro/helper that checks if a layer IS topdown/background instead?

Reply:
How would you tackle this down? If topdown layer usage for other planes will ever be implemented in the future, it'd still break a few more things this, and I've some doubts on how I could pave the way for it anyway, considering I'd still have to exclude stuff on the upper (topdown) layers, which is a bit of a unknown variable.